### PR TITLE
Fix image extraction or validation on build fails

### PR DESCRIPTION
**Pull request by kofrezo**
Commands to verify, download and extract image must have a exclusive
file lock and wait for each other. If not builds on the same hypervisor
might break each other e.g. when image gets downloaded, verfication
fails, images gets deleted and downloaded and while downloading another
build performs the same step.

I assume that 30 seconds is enough to download a image within a proper
network.

### CODE REVIEW

File: igvm/hypervisor.py [hunk 1]
```diff
@@ -316,8 +316,10 @@ def format_vm_storage(self, vm, transaction=None):
     def validate_image_checksum(self, image):
         """Compares the local image checksum against the checksum returned by
         foreman."""
+
+        image_file = '{}/{}'.format(IMAGE_PATH, image)
         local_hash = self.run(
-            'md5sum {}/{}'.format(IMAGE_PATH, image)
+            "flock -x -w 30 {} -c 'md5sum {}'".format(image_file, image_file)
```
Comment (HerrSpace) [inline igvm/hypervisor.py]: You can use {1} in both places and only pass image_file once.

Comment (kofrezo) [reply igvm/hypervisor.py]: {0} but I got the idea.

**Comment by HerrSpace** [comment]
Retest this please

**Comment by kofrezo** [comment]
Retest this please

**Comment by TuxPowered42** [comment]
https://github.com/innogames/igvm/pull/43 seems a better solution.

**Pull request closed without merging**